### DISCUSSION TITLE
Svelte and Cables

### DISCUSSION THREAD

**Discussion by cristianvogel** [cables-gl/cables_docs#510] [Answers & Questions] [answered]
Has anyone tried to embed a Cables sketch in a [Svelte](https://kit.svelte.dev/) or Svelte-Kit app? 

I saw that some of the other frameworks have had Cables sketch as component written by the team... would be great to be able to get Svelte in there too.

**Comment by karlosgliberal** [comment, marked as answer]
We have used it in this project:
https://github.com/karlosgliberal/bikolabs-stories
You can see it in production here https://riscanvi.bikolabs.io/
For the project we have created this component (it is a very improved first version)
https://github.com/karlosgliberal/bikolabs-historias/blob/main/src/components/Cables.svelte

From the buttons in svelte we modify a store. The wires component is reactive and when it sees a change it calls the wires functionTrigger operator to change the scene.

**Reply by cristianvogel** [reply]
Wonderful example , thankyou. The github link you posted is a 404 though. .... seems like it should be https://github.com/karlosgliberal/bikolabs-historias ?

**Reply by cristianvogel** [reply]
How did you get your editor to stop complaining about the CABLES object -- 'cannot find name' -- ?
<img width="300" alt="Screenshot 2023-03-30 at 09 34 21" src="https://user-images.githubusercontent.com/12829050/228763548-0cbeb014-21c4-4382-a30f-4b97f2045841.png">

**Reply by karlosgliberal** [reply]
Possibly you have some configuration in the associated editor some linter.
The Cables object will be loaded in execution when the patch is passed to it in the component and that is why it has no definition and that is why it gives an error.

**Reply by cristianvogel** [reply]
I have spent all day trying to work the Function.op ``` CABLES.patch.config.patchFunctiontrigger()``` this  to talk to the Cables patch, but CABLES.patch.config is always undefined... I see you got it working, which gives me hope..

**Reply by karlosgliberal** [reply]
I just checked that the 404 was because the url was translated: this is the correct one: https://github.com/karlosgliberal/bikolabs-historias

In static each directory is a cable patch downloaded with the CLI, this is the command: cables -e 1AiNBm -i -c -f /yourpath/

In /src/routes/+page.svelte is where we call the component with the patach (directory) parameter

I hope it works for you now
